nested-types.ipynb

```diff
@@ -0,0 +1,802 @@
+       "  ]"
+      ]
+     },
+     "execution_count": 31,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ty = pa.struct([pa.field('x', pa.string()),\n",
+    "                pa.field('y', pa.int8())])\n",
+    "column_4 = pa.array([(\"joe\", 1), (None, 2), None, (\"mark\", 4), (\"jane\", None)],\n",
+    "                    type=ty)\n",
+    "column_4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[<pyarrow.Buffer address=0x4bdc4021040 size=1 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020f80 size=1 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020e80 size=24 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020ec0 size=11 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020f00 size=1 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020f40 size=5 is_cpu=True is_mutable=True>]"
+      ]
+     },
+     "execution_count": 32,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_4.buffers()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<nanoarrow.c_lib.CArrayView>\n",
+       "- storage_type: 'struct'\n",
+       "- length: 5\n",
+       "- offset: 0\n",
+       "- null_count: 1\n",
+       "- buffers[1]:\n",
+       "  - validity <bool[1 b] 11011000>\n",
+       "- dictionary: NULL\n",
+       "- children[2]:\n",
+       "  - <nanoarrow.c_lib.CArrayView>\n",
+       "    - storage_type: 'string'\n",
+       "    - length: 5\n",
+       "    - offset: 0\n",
+       "    - null_count: 1\n",
+       "    - buffers[3]:\n",
+       "      - validity <bool[1 b] 10111000>\n",
+       "      - data_offset <int32[24 b] 0 3 3 3 7 11>\n",
+       "      - data <string[11 b] b'joemarkjane'>\n",
+       "    - dictionary: NULL\n",
+       "    - children[0]:\n",
+       "  - <nanoarrow.c_lib.CArrayView>\n",
+       "    - storage_type: 'int8'\n",
+       "    - length: 5\n",
+       "    - offset: 0\n",
+       "    - null_count: 1\n",
+       "    - buffers[2]:\n",
+       "      - validity <bool[1 b] 11110000>\n",
+       "      - data <int8[5 b] 1 2 0 4 0>\n",
+       "    - dictionary: NULL\n",
+       "    - children[0]:"
+      ]
+     },
+     "execution_count": 33,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "na.c_array_view(na.c_array(column_4))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Map\n",
+    "\n",
+    "Map type represents nested data where each value is a variable number of key-item pairs. Its physical representation is the same as a list of `{key, item}` structs.\n",
```

Comment:
It will be useful to explain the different with a struct, I think, as both logically represent key->value elements (certainly in a python they could both be thought of as an array of python dictionaries, which is how we convert them in to_pandas)